tap_github/sync.py

```diff
@@ -24,10 +24,52 @@ def get_selected_streams(catalog):
 
     return selected_streams
 
+def update_currently_syncing(state, stream_name):
+    """
+    Updates currently syncing stream in the state.
+    """
+    if stream_name is None and ('currently_syncing' in state):
+        del state['currently_syncing']
+    else:
+        singer.set_currently_syncing(state, stream_name)
+    singer.write_state(state)
+
+def update_currently_syncing_repo(state, repo_path):
+    """
+    Appends repository if completed syncing, 
+    and flushes `currently_syncing_repo` when all repositories are synced.
+    """
+    if (repo_path is None) and ('currently_syncing_repo' in state):
+        del state['currently_syncing_repo']
+    else:
+        state['currently_syncing_repo'] = repo_path
+    singer.write_state(state)
+
+def get_ordered_stream_list(currently_syncing):
+    """
+    Get an ordered list of remaining streams to sync followed by synced streams.
+    """
+    stream_list = list(sorted(STREAMS.keys()))
```

Comment:
sorting is not required as we already took sorted records.

Reply:
As `STREAMS` is a dictionary, it is not necessary that `.keys()` method will return a list of streams in the same order every time the tap will be synced. Hence, we need to sort the list.